homeassistant/components/hassio/issues.py

```diff
@@ -69,6 +83,88 @@
     "untrusted",
 }
 
+# Keys (type + context) of issues that when found should be made into a repair
+ISSUE_KEYS_FOR_REPAIRS = {
+    "issue_system_multiple_data_disks",
+    "issue_system_reboot_required",
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SuggestionDataType(TypedDict):
+    """Suggestion dictionary as received from supervisor."""
+
+    uuid: str
+    type: str
+    context: str
+    reference: str | None
+
+
+@dataclass(slots=True, frozen=True)
+class Suggestion:
+    """Suggestion from Supervisor which resolves an issue."""
+
+    uuid: str
+    type_: str
+    context: str
+    reference: str | None = None
+
+    @property
+    def key(self) -> str:
+        """Get key for suggestion (combination of context and type)."""
+        return f"{self.context}_{self.type_}"
+
+    @staticmethod
+    def from_dict(data: SuggestionDataType) -> Suggestion:
```

Comment:
This method can preferably be a `classmethod` as it returns an instance of the class.